atom/browser/api/atom_api_window.cc

```diff
@@ -587,9 +587,21 @@ void Window::SetFocusable(bool focusable) {
 void Window::SetProgressBar(double progress, mate::Arguments* args) {
   mate::Dictionary options;
   std::string mode;
+  NativeWindow::ProgressState state = NativeWindow::PROGRESS_NORMAL;
```

Comment:
The standard way is to add a `Converter` for the enum, so the type can be automatically converted:

```c++
NativeWindow::ProgressState mode;
options.Get("mode", &mode);
```

You can find an example of `ThumbarButton` in this file.